release-22.1: backupccl: parallelize loading of incremental manifests

adityamaru opened the pull request:
This change is a targetted change to parallelize the loading of backup manifests for each incremental layer of a backup in a SHOW BACKUP call.

This is a selective backport of #87599.

Release note: None

Release justification: performance improvement required for making SHOW BACKUP in the presence of many incremental layers more performant for CockroachCloud

adityamaru committed:
release-22.1: backupccl: parallelize loading of incremental manifests

This change is a targetted change to parallelize the loading
of backup manifests for each incremental layer of a backup
in a SHOW BACKUP call.

This is a selective backport of #87599.

Release note: None

Release justification: performance improvement required
for making SHOW BACKUP in the presence of many incremental layers
more performant for CockroachCloud

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93167)
<!-- Reviewable:end -->

adityamaru committed:
backupccl: adds an encrypted backup regression test

When we parallelized manifest handling in 22.2 we broke
encrypted backups. It seemed worthwhile to backport this test
to ensure we've done no such thing in the first commit here.

Release note: None
